Automattic/jetpack
# Connection: Add ConnectionErrorNotice React Component

#### Changes proposed in this Pull Request:
Add the `ConnectionErrorNotice` React component to display connection verified errors. This is part of a bigger project (1202697549597858-as) to provide consistent failed connection error reporting and an interface to reconnect (1202697549597858-as-1202697549597864). 

The changes will affect the following:
* Jetpack
* Connection

#### Other information:


#### Jetpack product discussion
* p9dueE-5w8-p2

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:
* Run `jetpack build packages/my-jetpack`.
* Enable the "Jetpack Debug Tools" plugin.
* Go to Jetpack Debug > Jetpack Debug.
* Enable "Broken token Utilities" and click "Save".
* Go to Jetpack Debug > Connection Errors
* Under "Create fake errors", select "With the blog token", select "Verified", and click the "Create error" button.
* Go to Jetpack > My Jetpack.
* You should see an error message as shown in the below screenshot:
![jetpack-connection-error-example](https://user-images.githubusercontent.com/56307/190708871-b9c573e3-0e90-451c-8fc0-7de2ebcd8d2e.png)
